api/storage_service.cc

```diff
@@ -644,6 +644,23 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         });
     });
 
+    ss::perform_keyspace_offstrategy_compaction.set(r, wrap_ks_cf(ctx, [] (http_context& ctx, std::unique_ptr<request> req, sstring keyspace, std::vector<sstring> tables) -> future<json::json_return_type> {
+        std::unordered_map<sstring, bool> needed;
+        for (const auto& t : tables) {
+            needed[t] = false;
+        }
+        co_await ctx.db.invoke_on_all([&keyspace, &tables, &needed] (replica::database& db) -> future<> {
+            for (const auto& table : tables) {
+                auto& t = db.find_column_family(keyspace, table);
+                auto needed_for_table = co_await t.perform_offstrategy_compaction();
```

Comment:
`needed |= co_await t.perform_offstrategy_compaction();`

Reply:
across shards?  I'm concerned it could end up assigning false, wiping out other shard's true value.
maybe unconditional assign of true from multiple shards if co_await t.perform_offstrategy_compaction()

Reply:
Right. We have ctx.db.map_reduce0(lambda_that_returns_bool, false, std::plus<>()) for that.

(adding two bools yields a bool with the correct semantics)

Reply:
fixed in v3